docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,520 @@
+/// thread, while the main thread calls `rclrs::spin()` to keep the node running.
+///
+/// The `new` function creates the node and the subscription, and returns a `SimpleSubscriptionNode`
+/// instance. The `data_callback` function can be used to access the latest received message.
+impl SimpleSubscriptionNode {
+    fn new(context: &Context) -> Result<Self, RclrsError> {
+        let node = create_node(context, "simple_subscription").unwrap();
+        let data: Arc<Mutex<Option<StringMsg>>> = Arc::new(Mutex::new(None));
+        let data_mut: Arc<Mutex<Option<StringMsg>>> = Arc::clone(&data);
+        let _subscriber = node
+            .create_subscription::<StringMsg, _>(
+                "publish_hello",
+                QOS_PROFILE_DEFAULT,
+                move |msg: StringMsg| {
+                    *data_mut.lock().unwrap() = Some(msg);
+                },
+            )
+            .unwrap();
+        Ok(Self {
+            node,
+            _subscriber,
+            data,
+        })
+    }
+    fn data_callback(&self) -> Result<(), RclrsError> {
+        if let Some(data) = self.data.lock().unwrap().as_ref() {
+            println!("{}", data.data);
+        } else {
+            println!("No message available yet.");
+        }
+        Ok(())
+    }
+}
+/// The `main` function creates a new ROS2 context, a `SimpleSubscriptionNode` instance, and starts a separate thread to periodically call the `data_callback` method on the subscription. The main thread then calls `rclrs::spin()` to keep the node running and receive messages.
+///
+/// The separate thread is used to ensure that the `data_callback` method is called regularly, even if the main thread is blocked in `rclrs::spin()`. This allows the subscriber to continuously process and print the received "hello" messages.
+fn main() -> Result<(), RclrsError> {
+    let context = Context::new(env::args()).unwrap();
+    let subscription = Arc::new(SimpleSubscriptionNode::new(&context).unwrap());
+    let subscription_other_thread = Arc::clone(&subscription);
+    thread::spawn(move || -> () {
+        iter::repeat(()).for_each(|()| {
+            thread::sleep(Duration::from_millis(1000));
+            subscription_other_thread.data_callback().unwrap()
+        });
+    });
+    rclrs::spin(subscription.node.clone())
+}
+```
+<details><summary>Examining the code in detail:</summary>
+
+#### The main Construct:
+```rust
+pub struct SimpleSubscriptionNode {
+    node: Arc<Node>,
+    _subscriber: Arc<Subscription<StringMsg>>,
+    data: Arc<Mutex<Option<StringMsg>>>,
+}
+```
+Instead of a Publisher, there is a Subscription object in the Subscriber node. The data needs to be an `Arc<Mutex<Option<StringMsg>>>` because there can be errors in the data transfer process and this can be caught by including the value of the incoming subscription in an optional.  
+#### This code defines a function named new that likely creates an instance of some SimpleSubscriptionNode.
```

Comment:
```suggestion
#### impl SimpleSubscriptionNode
This code defines methods for the `SimpleSubscriptionNode` `struct`. 

##### new
The `new` method creates a ROS 2 node, subscriber, and a storage location for received messages, storing them in the `struct`. 
```

Reply:
It has already been incorporated. What's the problem?

Reply:
I don't know, what's the problem? I committed this change along with the other suggestions that @maspe36 made